File: sample apps/python/app/shared/execute_around_command.py
import time
import uuid


def log_time(log_info):
    def partial(action, get_description):
        start = time.time()
        action()
        stop = time.time()
        duration = stop - start
        log_info("DURATION {} took {} milliseconds".format(get_description(), duration))

    return partial


def log_wrap(log_info):
    def partial(action, get_description):
        log_info("BEGIN {}".format(get_description()))
        action_time = log_time(log_info)
        action_time(action, get_description)
        log_info("END {}\n".format(get_description()))

    return partial


def handle_error(log_error, log_info):
    def partial(action, get_description):
        try:
            action_around = log_wrap(log_info)
            action_around(action, get_description)
        except ValueError as ex:
            correlation_id = str(uuid.uuid4())
            log_info("Correlation Id {}".format(correlation_id))
            log_info("FAILED {}\n".format(get_description()))
            log_error(ex)
            raise ValueError(
                "An error has occurred. Please contact support with correlation Id {}".format(correlation_id))

    return partial

File: sample apps/JavaScript/src/common/execute_around_command.js
import createUUID from './uuid';

export function logTime(logInfo) {
    return (action, getDescription) => {
        const start = new Date();
        action();
        const end = new Date();        
        const duration = Math.floor(end - start) / (1000);
        logInfo(`DURATION: ${getDescription()} took ${duration}`); 
    }
}

export function logWrap(logInfo) {
    return (action, getDescription) => {
        const action_timer = logTime(logInfo);
        logInfo(`BEGIN: ${getDescription()}`);
        action_timer(action, getDescription);
        logInfo(`END: ${getDescription()}`);
        logInfo('\n');
    }
}

export function handleError(logError, logInfo) {
    return (action, getDescription) => {
        const action_wrap = logWrap(logInfo);
        try {
            action_wrap(action, getDescription);
        } catch(ex) {
            const correlationId = createUUID();
            logError(ex);
            logInfo(`Correlation Id: ${correlationId}`);
            logInfo(`FAILED: ${getDescription()}`);
            logInfo('\n');
            throw new Error(`An error has occurred. Please contact support with correlation Id ${correlationId}`);
        }
    }
}